File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = { "a", "e", "i" ,"o" , "u"}
        count = 0
        l = 0
        maxc = 0

        for r in range(len(s)):
            if s[r] in vowels:
                count += 1

            if (r - l + 1) > k:
                if s[l] in vowels:
                    count -= 1
                l += 1
            maxc = max(maxc , count)
        return maxc
            





        

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        HashSet<Character> vowels = new HashSet<>(Arrays.asList('a' , 'e' , 'i' , 'o' , 'u')) ;
        int count = 0 , maxc = 0 , l = 0;

        for( int r = 0 ; r< s.length() ; r++)
        {
            if (vowels.contains(s.charAt(r)))
            {
                count ++ ;
            }
            if ((r - l + 1) > k)
            {
                if (vowels.contains(s.charAt(l)))
                {
                    count --;
                }
                l++ ;
            }
            maxc = Math.max(maxc , count) ;
        } 
        return maxc ;
        
    }
}